### PR TITLE
Add docs for RocketMQ pubsub

### LINKED ISSUE #2467: Add RocketMQ pubsub documentation
**What content needs to be created or modified?**

Add documentation for the RocketMQ pubsub component. It was added earlier in https://github.com/dapr/docs/issues/1403 and was removed later.

**Where should the new material be placed?**

Under supported-pubsubs

**The associated pull request from dapr/dapr, dapr/components-contrib, or other Dapr code repos**

https://github.com/dapr/dapr/pull/4675

**Additional context**

### PR DESCRIPTION
**Pull request by shubham1172**
Signed-off-by: Shubham Sharma <shubhash@microsoft.com>

Thank you for helping make the Dapr documentation better!

**Please follow this checklist before submitting:**
- [X] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://docs.dapr.io/contributing/contributing-overview/#developer-certificate-of-origin-signing-your-work))
- [X] [Read the contribution guide](https://docs.dapr.io/contributing/contributing-docs/)
- [ ] Commands include options for Linux, MacOS, and Windows within codetabs
- [X] New file and folder names are globally unique
- [X] Page references use shortcodes instead of markdown or URL links
- [ ] Images use HTML style and have alternative text
- [ ] Places where multiple code/command options are given have codetabs

In addition, please fill out the following to help reviewers understand this pull request:

## Description

This is a copy of the previous PR #1405 that was reverted.

## Issue reference

Closes #2467

### CODE REVIEW

File: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-alicloud-rocketmq.md [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+Follow the instructions [here](https://www.alibabacloud.com/help/doc-detail/200153.htm) to setup AliCloud RocketMQ
+  - Simplified Chinese translation [here](https://help.aliyun.com/document_detail/200153.html)
+
+## Per-call metadata fields
+
+### Partition Key
+
+When invoking the rocketmq pub/sub, its possible to provide an optional partition key by using the `metadata` query param in the request url.
+
+You need specifie `rocketmq-tag`,`"rocketmq-key"` in `metadata`
```
Comment (nyemade-uversky) [inline daprdocs/content/en/reference/components-reference/supported-pubsub/setup-alicloud-rocketmq.md]:
```suggestion
You need to specify `rocketmq-tag`,`"rocketmq-key"` in `metadata`
```

**Comment by nyemade-uversky** [comment]
@berndverst would you be able to give this a quick review to make sure the spec matches the implementation of this component? 

As part of this, would you be able to confirm whether the consumer ID is set to be the application ID by default.

File: daprdocs/content/en/reference/components-reference/supported-pubsub/_index.md [hunk 2]
```diff
@@ -32,6 +32,11 @@ Table captions:
 | [RabbitMQ]({{< ref setup-rabbitmq.md >}})             | Beta  | v1 | 1.6 |
 | [Redis Streams]({{< ref setup-redis-pubsub.md >}})    | Stable | v1 | 1.0 |
 
+### Alibaba Cloud
+| Name                                              | Status | Component version | Since |
+|---------------------------------------------------|--------| ---- |---------------|
+| [Alibaba Cloud RocketMQ]({{< ref setup-alicloud-rocketmq.md >}}) | Alpha | v1 | 1.8 |
```
Comment (msfussell) [inline daprdocs/content/en/reference/components-reference/supported-pubsub/_index.md]: This should not be under Alibaba. This should be under generic. Even the type says type: pubsub.rocketmq.

File: daprdocs/content/en/reference/components-reference/supported-pubsub/setup-alicloud-rocketmq.md [hunk 3]
```diff
@@ -0,0 +1,111 @@
+---
+type: docs
+title: "Alibaba Cloud RocketMQ"
```
Comment (msfussell) [inline daprdocs/content/en/reference/components-reference/supported-pubsub/setup-alicloud-rocketmq.md]: Reword to remove all references to Alibaba. This is a generic component. As per type `type: pubsub.rocketmq`

Comment (msfussell) [reply daprdocs/content/en/reference/components-reference/supported-pubsub/setup-alicloud-rocketmq.md]: This also means renaming this file to remove alibaba.

File: daprdocs/content/en/reference/components-reference/supported-pubsub/_index.md (hunk 2 above)
Comment (shubham1172) [reply daprdocs/content/en/reference/components-reference/supported-pubsub/_index.md]: Got it. This PR was approved earlier as an Alibaba specific component https://github.com/dapr/docs/pull/1405. Moving this to a generic one.

**Comment by shubham1172** [comment]
@msfussell, @nyemade-uversky I have updated this PR. This adds RocketMQ as a generic component. I also started digging into the actual implementation of the RocketMQ code and the original docs PR was not at all matching with the current implementation. I have updated the metadata specs and example. I also found a few issues with RocketMQ metadata which we will track with https://github.com/dapr/components-contrib/issues/1772.

**Comment by msfussell** [comment]
@shubham1172 - Great! So this PR still needs more work on the component code before updating here. I will mark this bug as wait on Code PR.

**Comment by shubham1172** [comment]
@msfussell the current docs are in-line with the functionality today. Once the code PR is checked in, the docs would also need update. Can we go ahead and merge this PR to start with?, and with the next code PR we can also expect a new docs PR.

**Review by msfussell — APPROVED** [review]
LGTM

**Pull request merged**